gateway/ln-gateway/src/client.rs

```diff
@@ -94,9 +96,24 @@ impl GatewayClientBuilder {
             client_builder.with_database(rocksdb);
         }
 
+        let client_secret = match client_builder
+            .load_decodable_client_secret::<[u8; 64]>()
+            .await
+        {
+            Ok(secret) => secret,
+            Err(_) => {
+                info!("Generating secret and writing to client storage");
+                let secret = PlainRootSecretStrategy::random(&mut thread_rng());
```

Comment:
**thought:** That argument to `random` is so needless (or rather: useful for 1% of the cases). We should probably have `random()` and `random_with(rng)`, where `random` just forwards to `random_with(&mut thread_rng())`

Reply:
At this point, the `RootSecretStrategy` trait is just for convenience and we even talked about possibly removing it or moving it to the `fedimint-cli` package. I don't wish to further iterate on that trait within this PR if that's ok with you.

Reply:
Fair.

Reply:
Yes, `RootSecretStrategy` feels like accidental complexity because I didn't think of just storing a `Vec<u8>` in the DB instead of `ClientSecret`, which then needed a way of specifying the encoding. Should likely be removed and the encoding/strategy can be baked into whatever manages the secret (not the client anymore).